### PR TITLE
Cypress tests setup

### LINKED ISSUE #15: Add client-side end-to-end testing
**This ticket is a...**
- [ ] Bug
- [ ] New feature
- [x] Dev QOL change
- [ ] Other?

**Description**
[Cypress](https://www.cypress.io/) is a popular choice for end-to-end testing. It does require some setup though, so this ticket is responsible for bootstrapping the project with Cypress and creating an example test.  The example test can just be a test to ensure the home page loads and the login button renders.

**Acceptance Criteria**
* Cypress tests can be run via npm script
* Sample test exists

### PR DESCRIPTION
**Pull request by windyScripts**
Closes #15 (replace with issue number)

# Description

Please include a summary of the changes. Please also include relevant motivation and context. List any dependencies that are required for this change.

Added a test based on cypress for the app being online with a rendered login button.

- Created a config file in root for cypress
- Placed cypress tests and related files in ./client/tests/cypress
- Added a script, cypress:open to open cypress.

Motivation: To implement front end testing with cypress.
Dependencies: Can be found at [the cypress installation documentation](https://docs.cypress.io/guides/getting-started/installing-cypress)

## Testing

Describe what a reviewer should do to test your code, including instructions for manually testing as well as relevant automated test suites to run.

install dependencies
start the app
run cypress:open
go to e2e from the menu
run the script cypress-base-test

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshots

NA

# Checklist:

- [x] Changes have new/updated automated tests, if applicable
- [ ] Changes have new/updated docs, if applicable
- [x] I have performed a self-review of my own code
- [x] I have added comments on any new, hard-to-understand code
- [x] Changes have been manually tested
- [x] Changes generate no new errors or warnings

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  specPattern:"./client/tests/cypress/e2e"
+},
+fixturesFolder: "./client/tests/cypress/fixtures",
+screenshotsFolder: "./client/tests/cypress/screenshots",
+videosFolder: "./client/tests/cypress/videos",
+downloadsFolder: "./client/tests/cypress/downloads",
+fileServerFolder:"./"
+});
+
+// For what the folders are: https://docs.cypress.io/guides/references/configuration#Folders-Files
```
Comment (timmyichen) [inline cypress.config.ts:17]: nit: newline

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,4 @@ node_modules
 .next
 .vscode
 dist
+cypress
```
Comment (timmyichen) [inline .gitignore:6]: nit: newline

**Review by timmyichen — CHANGES_REQUESTED** [review]
awesome! per the original issue can we also add a sample test? it can be as simple as "when i navigate to the home page, the login button exists"

**Comment by windyScripts** [comment]
Added the test file now.

File: .gitignore (hunk 2 above)
Comment (windyScripts) [reply .gitignore:6]: Added the test file. I'm not familiar with what nit: newline means. Should there be an empty line at the end of every  file?

**Comment by windyScripts** [comment]
I've managed this pull request badly, I'll be more attentive going forward.

**Pull request closed without merging**